crates/store/src/filesystem/utility.rs

```diff
@@ -0,0 +1,42 @@
+// SPDX-FileCopyrightText: 2022 Profian Inc. <opensource@profian.com>
+// SPDX-License-Identifier: AGPL-3.0-only
+
+use crate::Error;
+
+use cap_async_std::fs::DirEntry;
+use serde::Serialize;
+use sha2::{Digest, Sha256};
+
+pub const ALGORITHM_NAME: &str = "sha256";
+
+pub fn sha256(buffer: &[u8]) -> String {
+    let mut hash = Sha256::new();
+    hash.update(buffer);
+    hash.finalize()
+        .iter()
+        .map(|byte| format!("{:0x}", byte))
+        .collect()
+}
+
+pub fn encode_key<K>(key: K) -> Result<(Vec<u8>, String), Error>
+where
+    K: Serialize,
+{
+    let key_bincode = bincode::serialize(&key).map_err(Error::Bincode)?;
+    let identity = sha256(&key_bincode);
+    Ok((key_bincode, identity))
+}
+
+pub fn encode_t<T>(key: T) -> Result<Vec<u8>, Error>
```

Comment:
The name of this function does not convey the purpose of it. What are similar functions called in libstd or common libraries in ecosystem, e.g. serde?

Reply:
This function isn't used much so I can probably just get rid of it to make the code a bit more clear.